huaweicloud/services/ram/resource_huaweicloud_ram_resource_share.go

```diff
@@ -459,20 +479,40 @@ func resourceRAMShareDelete(ctx context.Context, d *schema.ResourceData, meta in
 	return nil
 }
 
-func disassociateRAMShare(client *golangsdk.ServiceClient, d *schema.ResourceData) error {
-	disassociateRAMShareHttpUrl := "v1/resource-shares/{resource_share_id}/disassociate"
-	disassociateRAMSharePath := client.Endpoint + disassociateRAMShareHttpUrl
-	disassociateRAMSharePath = strings.ReplaceAll(disassociateRAMSharePath, "{resource_share_id}", d.Id())
-	disassociateRAMShareOpt := golangsdk.RequestOpts{
+func associatePrincipalsAndResourceUrns(client *golangsdk.ServiceClient, d *schema.ResourceData, principals,
+	resourceUrns []interface{}) error {
+	associateHttpUrl := "v1/resource-shares/{resource_share_id}/associate"
+	associatePath := client.Endpoint + associateHttpUrl
+	associatePath = strings.ReplaceAll(associatePath, "{resource_share_id}", d.Id())
+	associateOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody: map[string]interface{}{
+			"principals":    principals,
+			"resource_urns": resourceUrns,
+		},
+	}
+	_, err := client.Request("POST", associatePath, &associateOpt)
+	if err != nil {
+		return fmt.Errorf("error associate RAM share principals and resource urns, %s", err)
+	}
+	return nil
+}
+
+func disassociatePrincipalsAndResourceUrns(client *golangsdk.ServiceClient, d *schema.ResourceData, principals,
```

Comment:
we can use the resource ID (whatever name we use...balabala) to replace the argument `d` (resource data)

Reply:
OK